### ISSUE TITLE
Bug: `one-var` `separateRequires` misidentifies shadowed `require` calls

### ISSUE THREAD

**Issue by sethamus** [eslint/eslint#21180]
### Environment

Node version:
npm version:
Local ESLint version:
Global ESLint version:
Operating System:

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
	{
		rules: {
			"one-var": [
				"error",
				{
					separateRequires: true,
					const: "always",
				},
			],
		},
	},
];
```
</details>

```js
function foo(require) {
	const bar = require("bar"), baz = "baz";
}
```

### What did you expect to happen?

No lint errors. The `require` function parameter shadows the global `require`, so this is not a CommonJS `require()` call.

### What actually happened?

```
Split requires to be separated into a single block.
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG9uZS12YXI6IFtcImVycm9yXCIsIHsgc2VwYXJhdGVSZXF1aXJlczogdHJ1ZSwgY29uc3Q6IFwiYWx3YXlzXCIgfV0gKi9cblxuZnVuY3Rpb24gZm9vKHJlcXVpcmUpIHtcblx0Y29uc3QgYmFyID0gcmVxdWlyZShcImJhclwiKSwgYmF6ID0gXCJiYXpcIjtcbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I agree that we should require that the `require` should resolve to a global.
Should we also require `sourceType: commonjs`?
Feel free to open a PR for this.

**Comment by sethamus** [comment]
> Should we also require sourceType: commonjs?

The rule is frozen, so I don't think we can make that change.
